projects/bprotocol/index.js

```diff
@@ -149,13 +153,58 @@
     return allLendingPlatformBalances;
   }
 
+  async function liquityTvl(timestamp, block) {
+    if (block < lFirstBlock) return { '0x5f98805A4E8be255a32880FDeC7F6728C6568bA0': '0' };
+
+    let totalBalance = new BigNumber(0);
+
+    for(let i = 0 ; ; i++) {
+      try {
+        const bamm = await sdk.api.abi.call(
+          {
+            block,
+            target: bKeeperAddress,
+            params: [i],
+            abi: abi["bamms"]
+          });
+        const balance = await sdk.api.abi.call(
+          {
+            block,
+            target: stabilityPoolAddress,
+            params: [bamm.output],
+            abi: abi["getCompoundedLUSDDeposit"]
+          });
+
+          totalBalance = totalBalance.plus(new BigNumber(balance.output));
+      }
+      catch {
+        break;
```

Comment:
Is there any way we can get the length of `bamms`? The current reliance on reverts makes the implicit assumption that RPC calls will not fail for other reasons other than revert, an assumption that hasn't held true in the past